metricbeat/templates/deployment.yaml

```
@@ -73,6 +73,10 @@ spec:
         image: "{{ .Values.image }}:{{ .Values.imageTag }}"
         imagePullPolicy: "{{ .Values.imagePullPolicy }}"
         args:
+        {{- if .Values.metricbeatConfig }}
```

Only checking for the top level key will cause this to fail for the scenario where someone is using the deprecated path with any other custom config that isn't `kube-state-metrics-metricbeat.yml`. For example if they were only modifying the `metricbeat.yml` which is used for the daemonset. Instead I think we should be explictly checking that this old path is being defined with something like: (not tested)


```suggestion
        {{- if .Values.metricbeatConfig.kube-state-metrics-metricbeat.yml }}
```